File: 05-04-2024_Make_The_String_Great.py
class Solution:
    def makeGood(self, s: str) -> str:
        ans = []
        for c in s:
            if ans and abs(ord(ans[-1]) - ord(c)) == 32:
                ans.pop()
            else:
                ans.append(c)
        return ''.join(ans)

File: 05-04-2024_Make_The_String_Great.cpp
class Solution {
public:
    string makeGood(string s) {
        int n=s.length();
        string ans="";
        for(int i=0;i<n;i++)
        {
            if(ans.empty())
            {
                ans.push_back(s[i]);
            }
            else if(!ans.empty()&&ans.back()!=s[i]&&tolower(ans.back())==tolower(s[i]))
            {
                ans.pop_back();
            }
            else
            {
                ans.push_back(s[i]);
            }
        }
        return ans;
    }
};
